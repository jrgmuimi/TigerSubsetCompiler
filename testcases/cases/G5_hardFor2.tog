let
    var loop_ctrl : int
    var random_var: int
in
    random_var := 5;
    for loop_ctrl:=0 to loop_ctrl+random_var do     // This is tricky 
        printint(loop_ctrl);
        printint(random_var);
        printstring("\n");
        random_var := random_var - 1;               // random_var is the actual loop control var, it -1 each iteration
        loop_ctrl := loop_ctrl - 1                  // manaully -1 from loop_ctrl, so loop_ctrl remains 0
    end
end