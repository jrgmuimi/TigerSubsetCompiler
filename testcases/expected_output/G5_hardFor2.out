MARS 4.5  Copyright 2003-2014 Pete Sanderson and Kenneth Vollmar

05
04
03
02
01
00

